src/internal/hooks/use-intersection-observer/__tests__/use-intersection-observer.test.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render } from '@testing-library/react';
+import { useIntersectionObserver } from '..';
+
+declare global {
+  interface Window {
+    IntersectionObserver: any;
+  }
+}
+
+function TestComponent() {
+  const { ref, isIntersecting } = useIntersectionObserver();
+  return <div ref={ref} data-testid="test" data-value={isIntersecting} />;
+}
+
+const mockObserve = jest.fn();
+const mockIntersectionObserver = jest.fn(() => ({
+  observe: mockObserve,
+  disconnect: jest.fn(),
+}));
+
+describe('useIntersectionObserver', () => {
+  beforeEach(() => {
+    window.IntersectionObserver = mockIntersectionObserver;
+  });
+
+  it('should create an observer with the defined target element', async () => {
+    const { findByTestId } = render(<TestComponent />);
+    const target = await findByTestId('test');
+
+    expect(mockIntersectionObserver).toHaveBeenCalled();
```

Comment:
As far as I can tell this relies on the test order, because `mockIntersectionObserver` and `mockObserve` are reused across these test cases. Currently this should work fine since the second test case does not check the usage of the mocks, but to make this test file more future-proof, I'd recommend to call `jest.clearAllMocks()` ([docs](https://jestjs.io/docs/jest-object#jestclearallmocks)) in the `beforeEach` handler.

Reply:
good catch!